modules/standard/Math.chpl

```diff
@@ -1048,6 +1048,72 @@ module Math {
     return chpl_gcd(x, y);
   }
 
+  /* Returns the greatest common divisor of the integer arguments `x` and
+     `y`. */
+  proc gcd(in x: int(32),in y: int(32)): int(32) {
+    (x, y) = (abs(x), abs(y));
+    while(y != 0) {
```

Comment:
Should this loop be in a generic helper function?

Reply:
Oops, I forgot about pure generic code.  I think the `(?w)` avoidance got me in a bad mindset